backend/src/main/java/fk/prof/backend/util/BitOperationUtil.java

```diff
@@ -0,0 +1,9 @@
+package fk.prof.backend.util;
+
+public class BitOperationUtil {
+
+  public static long constructLongFromInts(int mostSignificantBits, int leastSignificantBits) {
+    return (((long)mostSignificantBits) << 32) | (leastSignificantBits & 0xffffffffL);
```

Comment:
it'll work alright, but the value it turns up may not always be intuitive. Eg. long with value = 2^32 - 2 is very different from -1. You see what I mean?

There is nothing wrong with it inherently, its can be a little counter-intuitive if someone isn't thinking 2s complement representation while looking at the value.